service/worker/diagnostics/analytics/emitter.go

```diff
@@ -0,0 +1,74 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package analytics
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/uber/cadence/.gen/go/indexer"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/messaging"
+)
+
+type Emitter DataEmitter
+
+type emitter struct {
+	producer messaging.Producer
+}
+
+type NewEmitterParams struct {
+	Producer messaging.Producer
+}
+
+func NewEmitter(p NewEmitterParams) DataEmitter {
```

Comment:
do you see there will be a lot (>3) args to the `NewEmitter`. Is it worth introducing `NewEmitterParams` now?
in any way: nit - to rename `NewEmitterParams` to EmitterParams or something - it is not typical to call structure-holding-all-arguments as `NewSomething`.

Reply:
regarding more args, I am expecting logging and metrics.